packages/mask/src/plugins/CryptoartAI/SNSAdaptor/MakeOfferDialog.tsx

```diff
@@ -0,0 +1,272 @@
+import { useRemoteControlledDialog } from '@masknet/shared'
+import { WalletMessages } from '../../Wallet/messages'
+import type { useAsset } from '../hooks/useAsset'
+import { formatBalance, TransactionStateType } from '@masknet/web3-shared-evm'
+import { resolvePaymentTokensOnCryptoartAI } from '../pipes'
+import { usePlaceBidCallback } from '../hooks/usePlaceBidCallback'
+import { resolveAssetLinkOnCryptoartAI } from '../pipes'
+import { activatedSocialNetworkUI } from '../../../social-network'
+import { isTwitter } from '../../../social-network-adaptor/twitter.com/base'
+import { isFacebook } from '../../../social-network-adaptor/facebook.com/base'
+
+const useStyles = makeStyles()((theme) => {
+    return {
+        content: {
+            padding: 0,
+        },
+        details: {
+            color: '#999',
+            fontSize: '14px',
+            margin: '0px',
+        },
+        footer: {
+            display: 'flex',
+            justifyContent: 'flex-end',
+            padding: theme.spacing(0, 2, 2),
+        },
+        panel: {
+            marginTop: theme.spacing(2),
+            '&:first-child': {
+                marginTop: 0,
+            },
+        },
+        label: {},
+        buttons: {
+            width: '100%',
+            margin: `0 ${theme.spacing(-0.5)}`,
+        },
+        button: {
+            flex: 1,
+            margin: `${theme.spacing(1.5)} ${theme.spacing(0.5)} 0`,
+        },
+        markdown: {
+            margin: theme.spacing(1, 0),
+        },
+        mediaContent: {
+            display: 'flex',
+            justifyContent: 'center',
+            height: '200px',
+        },
+        player: {
+            maxWidth: '100%',
+            maxHeight: '100%',
+            border: 'none',
+        },
+    }
+})
+
+export interface MakeOfferDialogProps {
+    asset?: ReturnType<typeof useAsset>
+    open: boolean
+    onClose: () => void
+}
+
+export function MakeOfferDialog(props: MakeOfferDialogProps) {
+    const { asset, open, onClose } = props
+    const is24Auction = asset?.value?.is24Auction ?? false
+    const isVerified = (!asset?.value?.isSoldOut && !asset?.value?.is_owner) ?? false
+
+    const { t } = useI18N()
+    const { classes } = useStyles()
+
+    const chainId = useChainId()
+
+    const paymentTokens = resolvePaymentTokensOnCryptoartAI(chainId) ?? []
+    const selectedPaymentToken = first(paymentTokens)
+
+    const { amount, token, balance, setAmount, setToken } = useFungibleTokenWatched(selectedPaymentToken)
+
+    const [atLeastBidValue, setAtLeastBidValue] = useState(0)
+    useEffect(() => {
+        let atLeastBid = 0.01
+        if (asset?.value?.latestBidVo?.priceInEth) {
+            const price = new BigNumber(asset?.value?.latestBidVo.priceInEth)
+            atLeastBid = price.plus(price.gte(1) ? '0.1' : '0.01').toNumber()
+        }
+        setAtLeastBidValue(atLeastBid)
```

Comment:
```suggestion
        const price = new BigNumber(asset?.value?.latestBidVo.priceInEth)
        setAtLeastBidValue(price.isFinite() ? price.plus(price.gte(1) ? '0.1' : '0.01').toNumber() : 0.01)
```

see https://mikemcl.github.io/bignumber.js/#isF